### PR TITLE
ur-247: create flag icon component and add it to data transfer block

### PR DESCRIPTION
**Pull request by Djoudi21**
I added a flag component  and added it to the data transfer block...thanks to Jules haha

### CODE REVIEW

File: src/components/basic/FlagIcon.vue [hunk 1]
```diff
@@ -0,0 +1,28 @@
+<template>
+  <div v-if="country !== 'none'">
+    <img
+      class="h-6 w-12 rounded-sm"
+      :src="countryFlagSrc"
+      :alt="`Drapeau de la ${country}`"
```
Comment (Jabuf) [inline src/components/basic/FlagIcon.vue:6]: You should just use `${country}` instead.

File: src/components/basic/FlagIcon.vue [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+    />
+  </div>
+</template>
+<script setup>
+import { COUNTRIES_INFOS } from '@/utils/constants/countriesInfosConstant.js'
+import { computed } from 'vue'
+
+const props = defineProps({
+  country: {
```
Comment (Jabuf) [inline src/components/basic/FlagIcon.vue:15]: Maybe you should precise in the variable name what expected like `countryName` or `countryIso`.

File: src/utils/constants/countriesInfosConstant.js [hunk 3]
```diff
@@ -0,0 +1,2066 @@
+export const COUNTRIES_INFOS = {
```
Comment (Jabuf) [inline src/utils/constants/countriesInfosConstant.js:1]: We should make sure that the names are the same between our two files.

**Review by Jabuf — CHANGES_REQUESTED** [review]

File: src/components/basic/FlagIcon.vue (hunk 1 above)
Comment (prevostjules) [reply src/components/basic/FlagIcon.vue:6]: Yes, or use the $t to handle the translation (but I am not sure it is really required for now)

**Review by Jabuf — APPROVED** [review]
I've also added it to the DataCarrier block (which also had a chip with the country).
Otherwise this looks perfectly what we wanted, good job !

**Pull request merged**